frontend/src/app/utils/hooks.tsx

```diff
@@ -0,0 +1,25 @@
+import React, { useEffect } from "react";
+
+const useOutsideClick = (
+  ref: React.RefObject<HTMLDivElement>,
+  onClickOutside: () => void
+) => {
+  useEffect(() => {
+    const handleClickOutside = (ev: MouseEvent) => {
+      if (ref.current && !ref.current.contains(ev.target as Node)) {
+        onClickOutside();
+      }
+    };
+
+    document.addEventListener<"mousedown">("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener<"mousedown">(
+        "mousedown",
+        handleClickOutside
+      );
+    };
+    // eslint-disable-next-line
```

Comment:
Generally I don't like disabling dependency array warnings from React because we could end up with a reference to a stale dependency. Instead, I think we should add `onClickOutside` to the dependency array and then when using the hook just make sure to prevent the callback function from being recreated on each render using a `useCallback` hook.